lib/crypto/src/crypto.erl

```diff
@@ -642,87 +696,96 @@ nif_stub_error(Line) ->
 -ifndef(CRYPTO_VSN).
 -define(CRYPTO_VSN, "??").
 -endif.
+
 version() -> ?CRYPTO_VSN.
 
--spec start() -> ok | {error, Reason::term()}.
+-spec start() -> ok | {error, Reason :: term()}.
 start() ->
     case application:start(crypto) of
         ok ->
-            _ = supports(curves), % Build curves cache if needed
+            % Build curves cache if needed
+            _ = supports(curves),
             ok;
         Error ->
             Error
     end.
 
--spec stop() -> ok | {error, Reason::term()}.
+-spec stop() -> ok | {error, Reason :: term()}.
 stop() ->
     application:stop(crypto).
 
--spec supports() -> [Support]
-                        when Support :: {hashs,   Hashs}
-                                      | {ciphers, Ciphers}
-                                      | {public_keys, PKs}
-                                      | {macs,    Macs}
-                                      | {curves,  Curves}
-                                      | {rsa_opts, RSAopts},
-                             Hashs :: [sha1() | sha2() | sha3() | blake2() | ripemd160 | compatibility_only_hash()],
-                             Ciphers :: [cipher()],
-                             PKs :: [rsa | dss | ecdsa | dh | ecdh | eddh | ec_gf2m],
-                             Macs :: [hmac | cmac | poly1305],
-                             Curves :: [ec_named_curve() | edwards_curve_dh() | edwards_curve_ed()],
-                             RSAopts :: [rsa_sign_verify_opt() | rsa_opt()] .
+-spec supports() -> [Support] when
```

Comment:
I do prefer stuff like
```erlang
-spec supports(Input) -> Result
      when Input :: binary(),
           Result :: ok.
```
but maybe it's not "norm".

Reply:
That is very interesting.  Do you think you can find more examples, this could be worth exploring.

Reply:
I don't think I'll find many examples. This was the result I reached after a few iterations of working with 4-space tabs, tabs v. spaces, same line variables, etc., over the years.

Reply:
This is similar, for example: https://github.com/erlang/otp/blob/master/lib/stdlib/src/erl_eval.erl#L100, but not an exact match.

Reply:
The way @paulo-ferraz-oliveira prefer also gives nicer documentation if you use the feature to generate the xml from type specs.

Reply:
This is very interesting, but I am struggling to understand how the placement of `when` impacts the documentation.  Could you provide an example, maybe as a screen shot here or whatever is easiest?  Sorry for the inconvenience.

Reply:
So I tried to generate edocs using two different placements of when

```
-spec supports(Input) -> Result
   when Input :: binary(),
        Result :: ok.
supports(_) -> ok.

-spec supports1(Input) -> Result when
    Input :: binary(),
    Result :: ok.
supports1(_) -> ok.
```

But the edoc looks the same

![Screen Shot 2021-01-27 at 2 40 43 PM](https://user-images.githubusercontent.com/1829933/106006996-bbb8ef00-60ad-11eb-96a3-3bd91b92e0d8.png)

Were you using some other tool to generate docs?

Reply:
We are nearing finalizing our format, so it would be really helpful to hear what I am missing sooner rather than later.  Sorry for the short timeline, I know I have expected much more leniency for myself. @HansN @paulo-ferraz-oliveira

Reply:
@HansN @paulo-ferraz-oliveira am I using the wrong tool to generate these docs, what tool were you referring to?

Reply:
@richcarl do you maybe have an idea at the top of your head what is being referred to, to generate xml from type specs?